aws-transfer-user/src/main/java/software/amazon/transfer/user/CreateHandler.java

```diff
@@ -0,0 +1,125 @@
+package software.amazon.transfer.user;
+
+import static software.amazon.transfer.user.translators.Translator.translateToSdkHomeDirectoryMappings;
+import static software.amazon.transfer.user.translators.Translator.translateToSdkPosixProfile;
+import static software.amazon.transfer.user.translators.Translator.translateToSdkTags;
+
+import java.util.List;
+import java.util.Optional;
+import software.amazon.awssdk.services.transfer.TransferClient;
+import software.amazon.awssdk.services.transfer.model.CreateUserRequest;
+import software.amazon.awssdk.services.transfer.model.CreateUserResponse;
+import software.amazon.awssdk.services.transfer.model.HomeDirectoryType;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.transfer.user.translators.Translator;
+
+public class CreateHandler extends BaseHandlerStd {
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<TransferClient> proxyClient,
+            final Logger logger) {
+        this.logger = logger;
+
+        final String clientRequestToken = request.getClientRequestToken();
+        final ResourceModel newModel = request.getDesiredResourceState();
+
+        // Needed for tag updates
+        final String userArn = Translator.generateUserArn(request);
+        newModel.setArn(userArn);
+
+        final List<String> keysToAdd = translateToSShPublicKeyBodies(newModel);
+
+        return ProgressEvent.progress(newModel, callbackContext)
+                .then(progress -> createUser(proxy, proxyClient, clientRequestToken, progress))
+                .then(
+                        progress ->
+                                importSshPublicKeys(
+                                        proxy,
+                                        proxyClient,
+                                        clientRequestToken,
+                                        CREATE,
+                                        keysToAdd,
+                                        progress))
+                .then(
+                        progress ->
+                                addTags(
+                                        progress,
+                                        request,
+                                        newModel,
+                                        proxy,
+                                        proxyClient,
+                                        callbackContext))
+                .then(
+                        progress ->
+                                removeTags(
```

Comment:
How could there be any tags to remove when it's a new resource?

Reply:
Good point. Probably never triggers this code path.